library/cc/key_value_store.cc

```diff
@@ -0,0 +1,46 @@
+#include "library/cc/key_value_store.h"
+
+#include "library/common/data/utility.h"
+
+namespace Envoy {
+namespace Platform {
+
+namespace {
+
+envoy_data c_kv_store_read(envoy_data key, const void* context) {
```

Comment:
did you add `c_` prefix to avoid name conflicts? Can we document this?

Reply:
There will be no naming conflicts, due to the enclosing anonymous namespace. That said, this is our current convention for all platform implementations. For iOS this is called `ios_kv_store_read`, for Android, this is called `jni_kv_store_read`, and the `c_` prefix is already used elsewhere for the `cc` platform (arguably the prefix could/should be `cc_`, but it doesn't seem worth breaking the convention in this PR).  As for where to document this - I'm not sure? It's perhaps worth having a general documentation page on how platform callbacks work, and that's something I can start on.